api/scpca_portal/models/dataset.py

```diff
@@ -0,0 +1,73 @@
+import uuid
+
+from django.db import models
+
+from scpca_portal.models import APIToken, ComputedFile
+from scpca_portal.models.base import TimestampedModel
+
+
+class Dataset(TimestampedModel):
+    class Meta:
+        db_table = "dataset"
+        get_latest_by = "updated_at"
+        ordering = ["updated_at", "id"]
+
+    class FileFormats:
+        ANN_DATA = "ANN_DATA"
+        SINGLE_CELL_EXPERIMENT = "SINGLE_CELL_EXPERIMENT"
+
+        CHOICES = (
+            (ANN_DATA, "AnnData"),
+            (SINGLE_CELL_EXPERIMENT, "Single cell experiment"),
+        )
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    data = models.JSONField(default=dict)
+    format = models.TextField(choices=FileFormats.CHOICES, null=True)
+    email = models.EmailField(max_length=254, null=True)
+    regenerated_from = models.ForeignKey(
+        "self",
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="original_dataset",
+        help_text="Reference to the original dataset if regenerated",
+    )
+    start = models.BooleanField(
+        null=True,
+        help_text="Indicates if the dataset process has started.",
+    )
+
+    token = models.OneToOneField(
+        APIToken,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="dataset_token",
```

Comment:
To avoid a reverse query name clash, I’ve prefixed the `related_name` with "dataset_".